src/ff/curve_points.rs

```diff
@@ -0,0 +1,222 @@
+use curve25519_dalek::{
+    ristretto::{CompressedRistretto, RistrettoPoint},
+    Scalar,
+};
+use generic_array::GenericArray;
+use hkdf::Hkdf;
+use sha2::Sha256;
+use typenum::U32;
+
+use crate::{
+    ff::{ec_prime_field::Fp25519, Serializable},
+    secret_sharing::{Block, WeakSharedValue},
+};
+
+impl Block for CompressedRistretto {
+    type Size = U32;
+}
+
+///ristretto point for curve 25519
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub struct RP25519(CompressedRistretto);
+
+/// using compressed ristretto point
+impl WeakSharedValue for RP25519 {
+    type Storage = CompressedRistretto;
+    const BITS: u32 = 256;
+    const ZERO: Self = Self(CompressedRistretto([0_u8; 32]));
+}
+
+impl Serializable for RP25519 {
+    type Size = <<RP25519 as WeakSharedValue>::Storage as Block>::Size;
+
+    fn serialize(&self, buf: &mut GenericArray<u8, Self::Size>) {
+        *buf.as_mut() = self.0.to_bytes();
+    }
+
+    fn deserialize(buf: &GenericArray<u8, Self::Size>) -> Self {
+        RP25519(CompressedRistretto((*buf).into()))
```

Comment:
can we panic here instead of panicking on decompress?

Reply:
One reason why it may be useful to do is that ensures that data that enters our system is valid at the point of creation. Practically speaking it is much harder to debug if this system panics at the negation or multiplication. Validating assumptions at construction is generally considered good practice.

However the cost of this check is significant as you pointed out - decompressing a point is quite involved, so we could start with `debug_assert!(point.decompress().is_some())`